care/facility/models/events.py

```diff
@@ -0,0 +1,80 @@
+from django.contrib.auth import get_user_model
+from django.contrib.postgres.fields import ArrayField
+from django.db import models
+
+from care.utils.event_utils import CustomJSONEncoder
+
+User = get_user_model()
+
+
+class ChangeType(models.TextChoices):
+    CREATED = "CREATED"
+    UPDATED = "UPDATED"
+    DELETED = "DELETED"
+
+
+class EventType(models.Model):
+    parent = models.ForeignKey(
+        "self",
+        on_delete=models.PROTECT,
+        related_name="children",
+        null=True,
+    )
+    name = models.CharField(
+        max_length=100, null=False, blank=False, unique=True, db_index=True
+    )
+    description = models.TextField(null=True, blank=True)
+    model = models.CharField(max_length=50, db_index=True)
+    fields = ArrayField(models.CharField(max_length=50), default=list)
+    created_date = models.DateTimeField(auto_now_add=True)
+    is_active = models.BooleanField(default=True)
+
+    def get_descendants(self):
+        descendants = list(self.children.all())
+        for child in self.children.all():
+            descendants.extend(child.get_descendants())
+        return descendants
+
+    def save(self, *args, **kwargs):
+        if self.description is not None and not self.description.strip():
+            self.description = None
+        return super().save(*args, **kwargs)
+
+    def __str__(self) -> str:
+        return f"{self.model} - {self.name}"
+
+
+class PatientConsultationEvent(models.Model):
+    # external_id = use ulid
```

Comment:
We need some sort of non integer id for each event, i recommend making that the primary key